Python Examples/TestMonitor/results.py

```diff
@@ -0,0 +1,295 @@
+    :param parameters: The measurement parameters.
+    :param status:
+    :return: The step data used to create a test step.
+    """
+    step_status = status if status else {
+        "statusType": "RUNNING",
+        "statusName": "Running"
+        }
+
+    step_data = {
+        "stepId": None,
+        "parentId": None,
+        "resultId": None,
+        "children": None,
+        "data": parameters,
+        "dataModel": "TestStand",
+        "name": name,
+        "startedAt": None,
+        "status": step_status,
+        "stepType": step_type,
+        "totalTimeInSeconds": random.uniform(0, 1) * 10,
+        "inputs": inputs,
+        "outputs": outputs
+    }
+
+    return step_data
+
+
+def create_test_result_request(results):
+    """
+    Creates a create test result request object 
+    dictionary required for creating the new test results.
+    :param results: List of results that needs to create
+    :return: A dictionary which is required for creating the results
+    """
+    return {
+        "results":results
+    }
+
+
+def test_step_create_or_update_request_object(steps, update_result_total_time=True):
+    """
+    Creates a create/update test result request object 
+    dictionary required for creating the new/update existing test steps.
+    :param results: List of steps that needs to create/update
+    :param update_result_total_time: A boolean to state 
+    whether to update result total time or not
+    :return: A dictionary which is required for creating/updating the steps
+    """
+    return{
+        "steps": steps,
+        "updateResultTotalTime": update_result_total_time
+    }
+
+def update_test_results_request(results, determine_status_from_steps=True):
+    """
+    Creates a update test result request object 
+    dictionary required for updating the existing test results.
+    :param results: List of results that needs to be updated
+    :param determine_status_from_steps: A boolean representing 
+    whether the status of result should be updated based on result or not
+    :return: A dictionary which is required for updating the results
+    """
+    return{
+        "results": results,
+        "determineStatusFromSteps": determine_status_from_steps
+    }
+
+def print_usage_and_exit(error:str):
+    print("This example requires a configuration.")
+    print(error)
+    print()
+    print("Please specify a configuration using the following arguments:")
+    print()
+    print("\t <url> <api_key>")
+    print()
+    print("To run the example against a SystemLink Enterprise, the URL should include the")
+    print("scheme, host, and port if not default. For example:")
+    print("python <example_filename.py> https://myserver:9091 api_keynjnjnjnjnvgcycy")
+    quit()
+
+def main():
+    
+    args = sys.argv
+
+    if len(args) == 3 :
+        host =args[1]
+        api_key = args[2]
+    else:
+        print_usage_and_exit("Please pass all the required arguments")
```

Comment:
Instead of using `sys.argv` can we use `argparse` library so that we will be able to support `--help` argument? and also can we move this into a separate file outside as this can be reused for other API examples as well? This will help us avoid replicating the argument parsing and error handling.